cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/ConfigurationUtil.java

```diff
@@ -0,0 +1,112 @@
+package com.devonfw.cobigen.impl.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+
+import com.devonfw.cobigen.api.constants.ConfigurationConstants;
+import com.devonfw.cobigen.api.util.CobiGenPathUtil;
+
+/**
+ * Utilities related to the cobigen configurations including:
+ *
+ * 1. templates location
+ */
+public class ConfigurationUtil {
+
+    private static final Logger LOG = Logger.getLogger(ConfigurationUtil.class);
+
+    /**
+     * The method finds location of templates. It could be CobiGen_Templates folder or a template artifact
+     * @return template location uri if exist, otherwise null
+     */
+    public static URI findTemplatesLocation() {
+        Path configFile = getConfigurationFile();
+        if (configFile != null && Files.exists(configFile)) {
+            LOG.info("Configuration file is located at " + configFile);
```

Comment:
if you write log statements with additional data to be printed, please use log4j stringbuilder notation. In this case: `             LOG.info("Configuration file is located at {}", configFile);`
The difference here is, that string concatenation is already done before entering the info() method and thus this is lost computation power in case no INFO log should be written. By slf4j string builder notation, no computation is done at all before the decision is made by slf4j to print the log statement.